systemd/resolved/resolved.go

```diff
@@ -0,0 +1,169 @@
+// Copyright 2023 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package resolved
+
+import (
+	"context"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	"github.com/godbus/dbus/v5"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const namespace = "systemd_resolved"
```

Comment:
I think this can simply be `resolved`.

Reply:
I understand the point.

IMO, I think we should keep `systemd` even it is a bit verbose for following reasons.

- For  better searchability, users may start searching metrics from the word systemd.
- Consistency. To make it clear as a part of systemd exporter.
- To avoid conflicts with other similar daemons. So far, I think these systemd provided services don't directly conflict with other middleware.Though the naming is very common like resolved,networkd, timesyncd. I think we should keep systemd prefix to distinguish(and the service name is actually systemd-resolved).

What do you think?

Reply:
@jpds @SuperQ 
I think this naming policy will be applied for future collectors that gather metrics from systemd provided daemons, such as networkd, so please check the comment.
Ultimately, I will make changes as determined by the maintainers.

Reply:
:+1: , yes, it should be `systemd_resolved`.

Reply:
The other way to do this is to split namespace and subsystem for use with `prometheus.BuildFQName()`.

```go
const (
	namespace = "systemd"
	subsystem = "resolved"
)
```

Reply:
Thank you for quick comment!

Although the result is the same, to split namespace and subsystem seems to be meaningful to make the policy clear.

I'm going to make a small change for that.

Reply:
@SuperQ 
Sorry to bother you.
I've made a change for that: https://github.com/prometheus-community/systemd_exporter/pull/119/commits/625cf1140f8f8e75629ba2f8e555a62f63ae7eb7
and seems to be fine with metrics `systemd_resolved` prefixed.

Please take a look and finish the workflow.